collated/main/Ernest.md

```diff
@@ -0,0 +1,358 @@
+        }
+```
+###### \java\seedu\address\logic\commands\ListByBloodtypeCommand.java
+``` java
+/**
+ * Finds and lists all persons in address book whose blood type matches the keyword.
+ * Keyword matching is case insensitive.
+ */
+public class ListByBloodtypeCommand extends Command {
+
+    public static final String COMMAND_WORD = "bloodtype";
+    public static final String COMMAND_ALIAS = "bt"; // shorthand equivalent alias
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons of a blood type "
+            + " and displays them as a list with index numbers.\n"
+            + "Example: " + COMMAND_WORD + " AB+\n"
+            + "Example: " + COMMAND_ALIAS + " O";
+
+    private final BloodtypeContainsKeywordPredicate predicate;
+
+    public ListByBloodtypeCommand(BloodtypeContainsKeywordPredicate predicate) {
+        this.predicate = predicate;
+    }
+
+    @Override
+    public CommandResult execute() {
+        model.updateFilteredPersonList(predicate);
+        return new CommandResult(getMessageForPersonListShownSummary(model.getFilteredPersonList().size()));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof ListByBloodtypeCommand // instanceof handles nulls
+                && this.predicate.equals(((ListByBloodtypeCommand) other).predicate)); // state check
+    }
+}
+```
+###### \java\seedu\address\logic\commands\RelationshipCommand.java
+``` java
+/**
+  * Changes the relationship of an existing person in the address book.
+  */
+public class RelationshipCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "relation";
+    public static final String COMMAND_ALIAS = "rel";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the relationship of the person identified "
+            + "by the index number used in the last person listing. "
+            + "Existing relationship will be overwritten by the input.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + PREFIX_RELATIONSHIP + "[RELATIONSHIP]\n"
+            + "Example 1: " + COMMAND_WORD + " 1 "
+            + PREFIX_RELATIONSHIP + "John Doe"
+            + "Example 2: " + COMMAND_ALIAS + " 1 "
+            + PREFIX_RELATIONSHIP + "Mary Jane"
+            + "Example 3: " + COMMAND_WORD + " 1 "
+            + PREFIX_RELATIONSHIP;
+
+    public static final String MESSAGE_ADD_RELATIONSHIP_SUCCESS = "Added relationship to Person: %1$s";
+    public static final String MESSAGE_DELETE_RELATIONSHIP_SUCCESS = "Removed relationship from Person: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
+
+    private final Index index;
+    private final Relationship relation;
+
+    /**
+      * @param index of the person in the filtered person list to edit the relation
```

Comment:
The start of the header comment does not follow the convention (i.e. first sentence should be something like `Selects next line …` instead of `Select next line` or `This method is for selecting …`)